### ISSUE TITLE
Can't use an imported template as a pragma

### ISSUE THREAD

**Issue by deech** [nim-lang/Nim#14785]
### Example
```nim
# a.nim
template aPragma*() = ...
# b.nim
import a
proc b() {.aPragma.} = ...
```

I also tried"
```nim
proc b() {.a.aPragma.} = ...
```

### Current Output
```
invalid pragma: aPragma
```

### Expected Output
```
Successful compilation
```

```
$ nim -v
Nim Compiler Version 1.3.5 [Linux: amd64]
Compiled at 2020-06-20
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: 79c90b30ee5590b02f8bcfac68ed0c9ef7b38e7d
active boot switches: -d:release
```

**Comment by ghost** [comment]
@deech you need to annotate `aPragma` with {.pragma.} so it can be used as a pragma :)
```nim
# a.nim
template aPragma*() {.pragma.}
# b.nim
import a
proc b() {.aPragma.} = discard # works
```

**Comment by ghost** [comment]
See https://nim-lang.org/docs/manual.html#userminusdefined-pragmas-custom-annotations

**Comment by deech** [comment]
Yes, that works. As a side note if you want to use the template for metaprogramming and you leave out the initial argument that takes the body of the `proc` or you get the same error so you need to do:
```nim
template aPragma(body:untyped):untyped = body
proc p() {.aPragma.} = ...
```

Not totally related to my original issue but might be worth noting in the manual as well.

Another thing I noticed is qualified pragmas don't seem to work even if I have the `{.pragma.}` pragma, so `proc p() {.a.aPragma.}` fails with `identifier expected but found 'a.aPragma'`. Not sure if that's a bug but it surprised me so it might also be worth spelling out in the manual
 
In any case, thanks! This can be closed

**Comment by timotheecour** [comment]
it's a common error; i think we should improve the error message before closing
